test/spl/degree_elevation_test.cc

```diff
@@ -0,0 +1,516 @@
+  }
+
+ protected:
+  std::shared_ptr<spl::BSpline<1>> original_;
+  std::shared_ptr<spl::BSpline<1>> after_elevation;
+};
+
+TEST_F(Random1DBSplineForDegreeElevation, HasELevatedDegree) {  // NOLINT
+  ASSERT_THAT(after_elevation->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+}
+
+TEST_F(Random1DBSplineForDegreeElevation, HasMoreKnots) {  // NOLINT
+  ASSERT_THAT(after_elevation->GetKnotVector(0)->GetNumberOfDifferentKnots(),
+              original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation->GetKnotVector(0)->GetNumberOfKnots(),
+              original_->GetKnotVector(0)->GetNumberOfKnots()
+                  + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+}
+
+TEST_F(Random1DBSplineForDegreeElevation, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation->GetNumberOfControlPoints(),
+              original_->GetNumberOfControlPoints() + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+}
+
+TEST_F(Random1DBSplineForDegreeElevation, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation->AreGeometricallyEqual(*original_), true);
+}
+
+class Random1DNURBSForDegreeElevation : public Test {  // NOLINT
+ public:
+  Random1DNURBSForDegreeElevation() {
+    std::array<ParamCoord, 2> limits = {ParamCoord{0.0}, ParamCoord{1.0}};
+    spl::RandomNURBSGenerator<1> spline_generator(limits, 5, 3);
+    spl::NURBS<1> nurbs(spline_generator);
+    original_ = std::make_shared<spl::NURBS<1>>(nurbs);
+
+    spl::NURBS<1> elevation_spline(nurbs);
+    elevation_spline.ElevateDegree(0);
+    after_elevation_ = std::make_shared<spl::NURBS<1>>(elevation_spline);
+  }
+
+ protected:
+  std::shared_ptr<spl::NURBS<1>> original_;
+  std::shared_ptr<spl::NURBS<1>> after_elevation_;
+};
+
+TEST_F(Random1DNURBSForDegreeElevation, HasELevatedDegree) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+}
+
+TEST_F(Random1DNURBSForDegreeElevation, HasMoreKnots) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfDifferentKnots(),
+              original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfKnots(),
+              original_->GetKnotVector(0)->GetNumberOfKnots()
+                  + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+}
+
+TEST_F(Random1DNURBSForDegreeElevation, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetNumberOfControlPoints(),
+              original_->GetNumberOfControlPoints() + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+}
+
+TEST_F(Random1DNURBSForDegreeElevation, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation_->AreGeometricallyEqual(*original_, 0.01), true);
+}
+
+class A2DBSplineForDegreeElevation : public Test {  // NOLINT
```

Comment:
`A2DBSplineForDegreeElevationInDirection0`? Deviates from previous naming convection `ALinearNURBS`.

Reply:
I'm going to unify the naming.